### PR TITLE
Install extra APKs if received ( useful for Orchestrator )

### PR DESCRIPTION
**Pull request by CristianGM**
Now that we allow to use Orchestrator it would be nice to install the APKs required by Orchestrator to run.
Instead of being specific for Orchestrator this PR allows devs to install any extra APK they need.

~~There are no tests or documentation written yet because I want to validate the approach taken first, if it's ok I'll update the docs and add some tests ( suggestions on which ones are welcomed )~~

### CODE REVIEW

**Comment by CristianGM** [comment]
I made it more generic, now any APK used by tests: orchestrator, test-butler-app, ... can be installed

what do you think @jonas-m- ?
I think it does exactly what you suggested on your PR ( with the help of the plugin is just magic )

File: composer/src/main/kotlin/com/gojuno/composer/Main.kt [hunk 1]
```diff
@@ -112,8 +112,8 @@ private fun runAllTests(args: Args, testPackage: TestPackage.Valid, testRunner:
                     val installAppApk = device.installApk(pathToApk = args.appApkPath, timeout = installTimeout)
                     val installTestApk = device.installApk(pathToApk = args.testApkPath, timeout = installTimeout)
                     val installApks: MutableCollection<Observable<Unit>> = mutableListOf(installAppApk, installTestApk)
-                    if (args.runWithOrchestrator && args.orchestratorApks.isNotEmpty()) {
-                      installApks.addAll(args.orchestratorApks.map {
+                    if (args.extraApks.isNotEmpty()) {
```
Comment (koral--) [inline composer/src/main/kotlin/com/gojuno/composer/Main.kt]: Is this condition still needed?
It seems that if `extraApks` is empty then just nothing will be added in next line.

Comment (CristianGM) [reply composer/src/main/kotlin/com/gojuno/composer/Main.kt]: You are absolutely right. I'll remove it.

**Comment by CristianGM** [comment]
@yunikkk ?

**Comment by jonas-m-** [comment]
> I made it more generic, now any APK used by tests: orchestrator, test-butler-app, ... can be installed
> 
> what do you think @jonas-m- ?
> I think it does exactly what you suggested on your PR ( with the help of the plugin is just magic )

Yep - looks good - liking the generic approach :)

File: composer/src/main/kotlin/com/gojuno/composer/Main.kt [hunk 2]
```diff
@@ -111,9 +111,15 @@ private fun runAllTests(args: Args, testPackage: TestPackage.Valid, testRunner:
                     val installTimeout = Pair(args.installTimeoutSeconds, TimeUnit.SECONDS)
                     val installAppApk = device.installApk(pathToApk = args.appApkPath, timeout = installTimeout)
                     val installTestApk = device.installApk(pathToApk = args.testApkPath, timeout = installTimeout)
+                    val installApks: MutableCollection<Observable<Unit>> = mutableListOf(installAppApk, installTestApk)
```
Comment (trevjonez) [inline composer/src/main/kotlin/com/gojuno/composer/Main.kt]: nit: probably no need to specify type explicitly here?

File: README.md [hunk 3]
```diff
@@ -126,6 +126,11 @@ Composer shipped as jar, to run it you need JVM 1.8+: `java -jar composer-latest
   * Requires test orchestrator & test services APKs to be installed on device before executing.
   * More info: https://developer.android.com/training/testing/junit-runner#using-android-test-orchestrator
   * Example: `--with-orchestrator true`
+* `--extra-apks`
+  * Apks to be installed for utilities. What you would typically declare in gradle as `androidTestUtil` 
+  * Default: empty, only apk and test apk would be installed.
+  * Works great with orchestrator to install orchestrator & test services APKs.
```
Comment (artem-zinnatullin) [inline README.md]: nit: Orchestrator with capital O, can you please also make it a link to Android docs about it?

**Comment by trevjonez** [comment]
What machanism is used to install apks to the device? (I assume commander houses that functionality). I gave instant run another try this last week (still terrible) and noticed it used a single adb command to push all the apk files in one go. Not sure it would make any difference but might be worth looking at?

**Comment by artem-zinnatullin** [comment]
We use adb install if nothing changed. Instant run is not suitable for installs from scratch, it does push files and then uses tricks like custom classloader to hot swap to new stuff

File: README.md (hunk 3 above)
Comment (CristianGM) [reply README.md]: The link is just two lines before this, on the --with-orchestrator flag.

**Comment by artem-zinnatullin** [comment]
@yunikkk, @ming13 let's merge and release an update? I have rights to do a release, but not to merge

**Comment by yunikkk** [comment]
Guess ming13 added the merge permissions for you already...

**Comment by artem-zinnatullin** [comment]
Yep!

**Pull request merged**